reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -2392,7 +2415,7 @@ public int size() {
 
 		@Override
 		public String stepName() {
-			return "scalarSubscription(" + value + ")";
+			return stepName != null ? stepName : ("scalarSubscription(" + value + ")");
```

Comment:
see above comment, let's compute `stepName` upfront and make it final

Reply:
I did, but that would affect currect performance potentially as every scalar will calculate string used only in tests.

Reply:
mmh fair point, it might be slightly more acceptable to slow `stepName` than the constructor... keep in mind `stepName()` isn't necessarily only for tests, it's just that we don't have other prod code that depends on it, but it is very much for user consumption in both their prod and test code.

ok then, let's compute lazily here.

I think the field can still be final though, right?